Source/WebCore/platform/glib/PasteboardGLib.cpp

```diff
@@ -334,14 +356,33 @@ void Pasteboard::read(PasteboardFileReader& reader, std::optional<size_t> index)
 
     if (!index) {
         auto filePaths = platformStrategies()->pasteboardStrategy()->readFilePathsFromClipboard(m_name);
-        for (const auto& filePath : filePaths)
-            reader.readFilename(filePath);
+        if (filePaths.size() > 0) {
+            for (const auto& filePath : filePaths)
+                reader.readFilename(filePath);
+            return;
+        }
+    }
+
+    auto readBufferAtIndex = [&](size_t itemIndex) {
+        for (const auto& imageType : imageTypes()) {
+            if (!reader.shouldReadBuffer(imageType))
+                continue;
+            auto buffer = readBuffer(itemIndex, imageType);
+            if (buffer) {
```

Comment:
I should probably write this as:

```suggestion
            if (auto buffer = readBuffer(itemIndex, imageType)) {
```